File: examples/cs/request_reply/Reply.cs
using System;
using Dragonfly;


namespace Reply
{
    class Program
    {
        static void Main(string[] args)
        {
            bool KeepRunning = true;
            Module mod = new Module();

            string mm_ip = "localhost:7111";

            // Create a module and connect to Message Manager
            mod.ConnectToMMM(MID.REPLY, mm_ip);

            // Subscribe to messages
            mod.Subscribe(MT.REQUEST_TEST_DATA);
            mod.Subscribe(MT.EXIT);

            Console.WriteLine("Reply running...");

            Message m;
            int cnt = 1;
            while (KeepRunning)
            {
                try
                {
                    Console.WriteLine("\nWaiting for message");

                    m = mod.ReadMessage(Module.ReadType.Blocking);

                    if (m.msg_type == MT.REQUEST_TEST_DATA)
                    {
                        Console.WriteLine("Received message {0}", m.msg_type);

                        MDF.TEST_DATA data = new MDF.TEST_DATA();
                        data.a = cnt;
                        data.b = -17;
                        data.x = 234.456;
                        mod.SendMessage(MT.TEST_DATA, data);

                        Console.WriteLine("Sent message");
                        cnt++;
                    }
                    else if (m.msg_type == MT.EXIT)
                    {
                        KeepRunning = false;
                    }
                }
                catch
                {
                    Console.WriteLine("ERROR: Something went wrong...");
                    KeepRunning = false;
                }
            }

            mod.DisconnectFromMMM();
        }
    }
}

File: examples/matlab/request_reply/Reply.m
function Reply(mm_ip)

    global DF;

    Dragonfly_BaseDir = '../../..';
    addpath([Dragonfly_BaseDir '/lang/matlab']);

    Dragonfly_BaseDir

    MessageTypes =  { ...
                        'REQUEST_TEST_DATA' ...
                        'EXIT' ...
                    };

    ConnectArgs = {'REPLY', Dragonfly_BaseDir, ['./message_defs.mat']};
    if exist('mm_ip','var') && ~isempty(mm_ip)
        ConnectArgs{end+1} = ['-server_name ' mm_ip];
    end

    ConnectToMMM(ConnectArgs{:});
    Subscribe( MessageTypes{:})
    
    disp 'Reply running..'

    cnt = 1;
    while( 1)
        fprintf('\nWaiting for message\n');
        M = ReadMessage( 'blocking');
        
        switch(M.msg_type)
            case 'REQUEST_TEST_DATA'
                fprintf('Received message %s\n', M.msg_type);
            
                msg = DF.MDF.TEST_DATA;
                msg.a = int32(cnt);
                msg.b = int32(-30);
                msg.x = 123.456;
                SendMessage( 'TEST_DATA', msg);
                
                fprintf('Sent reply\n');
                cnt = cnt + 1;
            case 'EXIT'
                break; 

        end

    end

    DisconnectFromMMM

